src/main/java/io/airlift/units/DataSize.java

```diff
@@ -45,7 +45,7 @@ public static DataSize succinctBytes(long bytes)
 
     public static DataSize succinctDataSize(double size, Unit unit)
     {
-        return new DataSize(size, unit).convertToMostSuccinctDataSize();
+        return convertToMostSuccinctDataSize(size, unit);
```

Comment:
Have you measured the effect of this change? The VM should be able to elide this allocation, so I'm wondering if you're seeing some other effect, instead. Also, how did you measure the allocations?